src/components/v2/appDetails/appDetails.store.ts

```diff
@@ -0,0 +1,138 @@
+import { BehaviorSubject } from 'rxjs';
+import { ApplicationObject, iNode } from "./appDetails.type";
+import { URLS } from "../../../config";
+
+let applicationObjectTabsSubject: BehaviorSubject<Array<ApplicationObject>> = new BehaviorSubject([] as Array<ApplicationObject>);
+let _nodeTreeActiveParentNode: iNode
+let _nodeTreeActiveNode: iNode
+let _maxTabAllowd = 6
+
+const addAOT = (tabName: string, tabUrl: string, isSelected: boolean, title?: string) => {
+    let tab = {} as ApplicationObject
+    tab.name = tabName
+    tab.url = tabUrl
+    tab.isSelected = isSelected
+    tab.title = title || tabName
+    return tab
+}
+
+export const AppDetailsTabs = {
+    k8s_Resources: "K8s Resources",
+    log_analyzer: "Log Analyzer"
+}
+
+const AppDetailsStore = {
+    getAppDetailsTabs: () => {
+        return applicationObjectTabsSubject.getValue()
+    },
+    getAppDetailsTabsObservable: () => {
+        return applicationObjectTabsSubject.asObservable()
+    },
+    initAppDetailsTabs: (_url: string, displayLogAnalyzer: boolean, isLogAnalyserURL: boolean) => {
+        let aots = [] as Array<ApplicationObject>
+
+        aots.push(addAOT(AppDetailsTabs.k8s_Resources, _url + "/" + URLS.APP_DETAILS_K8, !isLogAnalyserURL))
+
+        if (displayLogAnalyzer) {
+            aots.push(addAOT(AppDetailsTabs.log_analyzer, _url + "/" + URLS.APP_DETAILS_LOG, isLogAnalyserURL))
+            _maxTabAllowd = 7
+        }
+
+        applicationObjectTabsSubject.next([...aots])
+
+        _nodeTreeActiveParentNode = undefined 
+        _nodeTreeActiveNode = undefined
+    },
+    addAppDetailsTab: (tabKind: string, tabName: string, tabURL: string) => {
+
+        if (!tabName || !tabURL || !tabKind) return
+
+        let applicationObjectTabs = applicationObjectTabsSubject.getValue()
+
+        if (applicationObjectTabs.length === _maxTabAllowd) {
+            return false
+        }
+
+        let alredyAdded = false
+        let title = tabKind + '/' + tabName
+        tabName = tabKind + '/...' + tabName.slice(-6)
+
+        for (let index = 0; index < applicationObjectTabs.length; index++) {
+            const tab = applicationObjectTabs[index];
+            tab.isSelected = false
+            if (tab.name.toLowerCase() === tabName.toLowerCase()) {
```

Comment:
logic of isSelected is different at different places, we should use same logic across or rather use this isSelected flag everywhere

Reply:
also  different objects can have same name, unique attribute is kind/name